src/data.ts

```diff
@@ -63,4 +63,20 @@ export class STData<ValueT> {
 	public reset(): void {
 		this.value = null;
 	}
+
+	public add(amt: number): void {
+		if (typeof amt !== 'number') {
+			return;
+		}
+
+		const val = this.getNull();
+
+		if (typeof val !== 'number') {
+			return;
+		}
+
+		const setVal = val + amt;
+
+		this.set(setVal as any);
```

Comment:
Using `as any` effectively disables type checks and should be avoided when possible. Based on the type checks for `amt` and `val` above, it shouldn't be possible for `setVal` to be anything other than a number. It could possibly be `NaN` or infinite, but that's out of scope for this function.

Is it possible to change `this.set(setVal as any)` to `this.set(setVal)` ? What happens?